content/operate/kubernetes/security/vault.md

```diff
@@ -0,0 +1,530 @@
+           --namespace <K8S_NAMESPACE> \
+           --from-file=vault.ca=<vault-ca-cert-file-path>
+   ```
+
+   {{<warning>}}
+   The Vault server certificate must be signed by the Certificate Authority provided in this secret.
+   {{</warning>}}
+
+## Create Redis Enterprise clusters
+
+1. Generate cluster credentials
+
+   Unlike standard deployments, Vault integration requires manually creating cluster credentials:
+
+   ```bash
+   # Generate a secure random password
+   openssl rand -base64 32
+   ```
+
+   Store credentials in Vault:
+
+   ```bash
+   vault kv put -namespace=<VAULT_NAMESPACE> \
+     <VAULT_SECRET_ROOT>/redisenterprise-<K8S_NAMESPACE>/<REC_NAME> \
+     username=<YOUR_USERNAME> \
+     password=<YOUR_PASSWORD>
+   ```
+
+   {{< alert title="Important notes" >}}
+   - The username field in the REC spec is ignored when using Vault
+   - The username from the Vault secret takes precedence
+   - Use strong, unique passwords for each cluster
+   {{</alert>}}
+
+   <br>
+
+2. Create cluster service account role
+
+   Configure a Vault role for the Redis Enterprise cluster's service account:
+
+   ```bash
+   vault write -namespace=<VAULT_NAMESPACE> \
+     auth/<AUTH_PATH>/role/redis-enterprise-rec-<K8S_NAMESPACE> \
+     bound_service_account_names=<REC_NAME> \
+     bound_service_account_namespaces=<K8S_NAMESPACE> \
+     policies=redisenterprise-<K8S_NAMESPACE>
+   ```
+
+   <br>
+
+3. Deploy Redis Enterprise cluster
+
+   Create the `RedisEnterpriseCluster` resource with Vault configuration:
+
+   ```yaml
+   apiVersion: app.redislabs.com/v1
+   kind: RedisEnterpriseCluster
+   metadata:
+     name: rec
+     namespace: <K8S_NAMESPACE>
+     labels:
+       app: redis-enterprise
+   spec:
+     nodes: 3
+     clusterCredentialSecretName: rec
+     clusterCredentialSecretType: vault
+     clusterCredentialSecretRole: redis-enterprise-rec-<K8S_NAMESPACE>
+     vaultCASecret: vault-ca-cert
+     podAnnotations:
+       vault.hashicorp.com/auth-path: auth/<AUTH_PATH>
+       vault.hashicorp.com/namespace: <VAULT_NAMESPACE>
+   ```
+
+   Apply the configuration:
+
+   ```bash
+   kubectl apply -f redis-enterprise-cluster.yaml
+   ```
+
+   Key configuration fields:
+
+   | Field | Description | Example |
+   |-------|-------------|---------|
+   | `clusterCredentialSecretName` | Name of the secret in Vault containing cluster credentials | `rec` |
+   | `clusterCredentialSecretType` | Must be set to `vault` | `vault` |
+   | `clusterCredentialSecretRole` | Vault role for cluster authentication | `redis-enterprise-rec-<K8S_NAMESPACE>` |
+   | `vaultCASecret` | Kubernetes secret containing Vault's CA certificate | `vault-ca-cert` |
+   | `podAnnotations` | Vault agent annotations for pod-level configuration | See example above |
+
+
+
+## Create Redis Enterprise databases
+
+To create a Redis Enterprise database (REDB) with Vault integration:
+
+1. Create database password in Vault:
+   ```bash
```

Comment:
Same with this code block:
<img width="670" height="227" alt="Screenshot 2025-11-14 at 14 18 28" src="https://github.com/user-attachments/assets/8929c394-4bd4-488c-b8b2-e93ab2c79540" />